src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -944,23 +946,65 @@ void IHttp2StreamLifetimeHandler.OnStreamCompleted(int streamId)
             }
         }
 
+        // We can't throw a Http2StreamErrorException here, it interrupts the header decompression state and may corrupt subsequent header frames on other streams.
+        // For now these either need to be connection errors or BadRequests. If we want to downgrade any of them to stream errors later then we need to
+        // rework the flow so that the remaining headers are drained and the decompression state is maintained.
         public void OnHeader(Span<byte> name, Span<byte> value)
         {
+            // https://tools.ietf.org/html/rfc7540#section-6.5.2
+            // "The value is based on the uncompressed size of header fields, including the length of the name and value in octets plus an overhead of 32 octets for each header field.";
+            _totalParsedHeaderSize += 32 + name.Length + value.Length;
```

Comment:
nit: `HeaderField.RfcOverhead`? I would prefer if this was a named constant instead.